opentelemetry-otlp/tests/integration_test/src/metrics_asserter.rs

```diff
@@ -1,40 +1,64 @@
+use anyhow::Result;
+use serde_json::Value;
 use std::fs::File;
+use std::io::{BufReader, Read};
 
-use opentelemetry_proto::tonic::metrics::v1::{MetricsData, ResourceMetrics};
+pub fn read_metrics_from_json(file: File) -> Result<Value> {
+    // Create a buffered reader for the file
+    let mut reader = BufReader::new(file);
+    let mut contents = String::new();
+
+    // Read the file contents into a string
+    reader
+        .read_to_string(&mut contents)
+        .expect("Failed to read json file");
+
+    // Parse the contents into a JSON Value
+    let metrics_data: Value = serde_json::from_str(&contents)?;
+    Ok(metrics_data)
+}
 
 pub struct MetricsAsserter {
-    results: Vec<ResourceMetrics>,
-    expected: Vec<ResourceMetrics>,
+    results: Value,
+    expected: Value,
```

Comment:
The earlier code was explicitly testing the conversion of JSON to proto structs, while it seems the current code no longer does this?

Reply:
The [grpc_build.rs](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-proto/tests/grpc_build.rs) adds the annotations for serialization/deserialization between json and proto structs, and this got tested in current setup.

Reply:
> The earlier code was also testing the conversion of JSON to proto structs, while it seems the current code no longer does this?

I changed it because the roundtrip for metrics is broken and its a can of worms - [this test I added as `ignore`](https://github.com/open-telemetry/opentelemetry-rust/pull/2432/files#diff-60ed943e21eb7847d83c35cae40bdf2ae03d82942c3aa7f15190174145e32cc3R157-R170) demonstrates it. I will raise a separate issue later to track this. Some of the fields drop out when you roundtrip, including e.g. the value. This is happening on the deserialization side, not the serialization side, which I think is less of a worry for us!

I had a bit of a look but because it is all tied up into the protobuff and serde serialization magic I didn't want to pull that into this PR.

Reply:
do we need to test that?

Reply:
I am totally fine to just get an ACK from Collector that it accepted our metrics/logs/traces. That will give instant value (to validate lot of stuff we are manually validating now, like is the shutdown going to panic or do its job etc.!)

Validating the actual content - Its important, but I consider it as non-blocking now, and can be added later too.

Reply:
Just to be clear: we're getting an ACK from the collector, and parsing all the results back out from the copy written by the collector to its file outputs, and comparing them to our expectation - we're validating the data that comes out of the collector is what we sent it now ✅ 

The only "thing" is, we're just using Serde models to deserialize for that validation, not our own proto-derived ones, because of the aforementioned fields-going-missing issue that would make a proto-based-deserialization  test succeed when data is actually lost.

So - i think this is in a pretty good state 🙏 !

Reply:
> do we need to test that?

Yes it's important to test from _opentelemetry-proto_ prospective that we can successfully deserialize the metrics data written by collector. This is a standalone crate, and also being used by users for consumers for collector. And the only way to test it is with integration tests. This is not the blocker, but we should bring it back once the serde model is fixed for metrics.

Reply:
Please add a TODO (with reference to the issue), so that we know this would be fixed eventually.